### PR TITLE
feat(#253): remove requirement that word be in list for repetition

### PR DESCRIPTION
**Pull request by grantlemons**
There are some cases where words are repeated legitimately, but I feel these are rare enough that a false-positive isn't that bad.

~~The more interesting case is when homographs appear next to each other, I don't know of any way we can correct for these, but here are some examples ([source](https://english.stackexchange.com/a/37835)):~~

~~- to record record profits~~
~~- to address address problems~~
~~- to refuse refuse charges~~

### CODE REVIEW

**Comment by mcecode** [comment]
Just a question since I saw the VSCode smoke test failed, so I can update it if needed. Is the change in the span length/range returned intended? For example, in the sentence "I like like this." It now seems to return the range from the space after the first like to the end of the second like (" like") instead of the whole "like like" previously.

**Comment by grantlemons** [comment]
@mcecode My starting point for this change was to rollback to before we used patterns for this lint, which apparently didn't have the same span behavior as the most recent version.
I've changed it such that the matching span should cover both words.

File: harper-core/src/word_metadata.rs [hunk 1]
```diff
@@ -113,6 +113,41 @@ impl WordMetadata {
         )
     }
 
+    /// Checks if the word is a homophone
+    /// This is based off of metadata, so it is not guaranteed to be correct
+    pub fn is_homophone(&self) -> bool {
+        [
```
Comment (elijah-potter) [inline harper-core/src/word_metadata.rs:119]: I'd like to backtrack something I said earlier. Change the name of this function to `is_likely_homophone` to indicate that, unlike the other `is_&` functions, this is not definitive.

File: harper-core/src/word_metadata.rs [hunk 2]
```diff
@@ -113,6 +113,41 @@ impl WordMetadata {
...
+                    is_plural: None,
+                    is_possessive: None,
+                    is_pronoun: None
+                })
+            ),
+            matches!(
+                self.verb,
+                Some(VerbData {
+                    is_linking: None,
+                    tense: None
```
Comment (elijah-potter) [inline harper-core/src/word_metadata.rs:133]: It feels like implementation is likely to result in an inconsistency with the model for `WordMetadata`. Is there a way to make this throw an error if we modify `WordMetadata` but don't update the code here?

File: harper-core/src/linting/repeated_words.rs [hunk 3]
```diff
@@ -71,6 +47,12 @@ mod tests {
...
     #[test]
     fn catches_basic() {
-        assert_lint_count("I wanted the the banana.", RepeatedWords::new(), 1)
+        assert_lint_count("I wanted the the banana.", RepeatedWords::default(), 1)
+    }
+
+    #[test]
+    fn does_not_lint_homophones() {
+        assert_lint_count("To address address problems.", RepeatedWords::default(), 0);
+        assert_lint_count("To record record profits.", RepeatedWords::default(), 0);
```
Comment (elijah-potter) [inline harper-core/src/linting/repeated_words.rs:56]: Make each of these lines a separate test.

File: harper-core/src/linting/repeated_words.rs [hunk 4]
```diff
@@ -1,66 +1,42 @@
...
+pub struct RepeatedWords;
+
+impl Linter for RepeatedWords {
+    fn lint(&mut self, document: &Document) -> Vec<Lint> {
+        let mut lints = Vec::new();
+
+        for chunk in document.iter_chunks() {
+            let mut iter = chunk.iter_word_indices().zip(chunk.iter_words()).peekable();
+
+            while let (Some((idx_a, tok_a)), Some((idx_b, tok_b))) = (iter.next(), iter.peek()) {
```
Comment (elijah-potter) [inline harper-core/src/linting/repeated_words.rs:15]: Love it.

**Review by elijah-potter — CHANGES_REQUESTED** [review]
Looks good, just some bookkeeping and future-proofing needed before merging.

File: harper-core/src/word_metadata.rs (hunk 1 above)
Comment (elijah-potter) [reply harper-core/src/word_metadata.rs:119]: I would also like to point out that these are homographs, not homophones

File: harper-core/src/word_metadata.rs (hunk 2 above)
Comment (grantlemons) [reply harper-core/src/word_metadata.rs:133]: Hmm, not sure about that. I can't think of any way.

Comment (elijah-potter) [reply harper-core/src/word_metadata.rs]: I found a way. We're good here.

**Review by elijah-potter — APPROVED** [review]

**Pull request merged**